lxd/api_cluster.go

```diff
@@ -3009,6 +3016,53 @@ func clusterNodeStatePost(d *Daemon, r *http.Request) response.Response {
 	return response.BadRequest(fmt.Errorf("Unknown action %q", req.Action))
 }
 
+func internalClusterHeal(d *Daemon, r *http.Request) response.Response {
+	migrateFunc := func(s *state.State, r *http.Request, inst instance.Instance, targetMemberInfo *db.NodeInfo, live bool, startInstance bool, metadata map[string]any, op *operations.Operation) error {
+		// This returns an error if the instance's storage pool is local.
+		// Since we only care about remote backed instances, this can be ignored and return nil instead.
+		poolName, err := inst.StoragePool()
+		if err != nil {
```

Comment:
Is the error that is returned `api.StatusErrorf(http.StatusNotFound, "Instance storage pool not found")`?

If so we should test for that here in order to catch other errors, using:

```go
poolName, err := inst.StoragePool()
if err != nil {
    if api.StatusErrorCheck(err, http.StatusNotFound) {
        return nil // We only care about remote backed instances.
    }

    return err
}
```

Reply:
> Is the error that is returned api.StatusErrorf(http.StatusNotFound, "Instance storage pool not found")?

Yes, that's the error I got.